.github/workflows/coverage.yml

```diff
@@ -12,25 +12,25 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-go@v4
         with:
-          go-version: "1.22.6"
+          go-version: "1.23.1"
       - name: Create a file with all the pkgs
         run: go list ./... > pkgs.txt
       - name: Split pkgs into 4 files
         run: split -d -n l/4 pkgs.txt pkgs.txt.part.
       # cache multiple
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
```

Comment:
Just curious to know whether this is also necessary for the Go version upgrade to work? or it is an independent change?

Reply:
All are necessary because they don't support the new version of go, unless I am missing something